### PR TITLE
Add filePattern parameter to the file move operation

### PR DESCRIPTION
**Pull request by sajinieKavindya**
Add filePattern parameter to the file move operation. 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java [hunk 1]
```diff
@@ -222,14 +235,14 @@ private boolean moveFile(FileObject srcFile, boolean createNonExistingParents,
      * any file.
      *
      * @param srcFile                  Src directory to move
-     * @param createNonExistingParents True if to create directories along the path if not exist
      * @param destinationFile          Destination folder
      * @param overWrite                True if to overwrite any existing file when moving
+     * @param filePattern              The pattern (regex) of the files to be moved
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java:240]: Add javadoc for isSuccessful parameter

File: src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java [hunk 2]
```diff
@@ -239,26 +252,73 @@ private boolean moveFolder(FileObject srcFile, boolean createNonExistingParents,
...
+                boolean result;
+                if (child.getType() == FileType.FILE) {
+                    result = moveFileWithPattern(child, destinationFile, filePattern);
+                } else if (child.getType() == FileType.FOLDER) {
+                    String newDestination = destinationFile.getPublicURIString() + Const.FILE_SEPARATOR
+                            + child.getName().getBaseName();
+                    result = moveFolder(child, fsManager.resolveFile(newDestination, fso),
+                            overWrite, filePattern, isSuccessful);
+                } else {
+                    log.error("Could not move the file: " + child.getName() + "Unsupported file type: "
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java:285]:
```suggestion
                    log.error("Could not move the file: " + child.getName() + ", Unsupported file type: "
```

File: src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java [hunk 3]
```diff
@@ -239,26 +252,73 @@ private boolean moveFolder(FileObject srcFile, boolean createNonExistingParents,
...
+        } else {
+            srcFile.moveTo(destinationFile);
         }
+        return isSuccessful;
+    }
 
-        srcFile.moveTo(destinationFile);
+    /**
+     * @param remoteFile               Location of the file
+     * @param target              New file location
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java:304]: Not aligned with other comments

File: src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java [hunk 4]
```diff
@@ -239,26 +252,73 @@ private boolean moveFolder(FileObject srcFile, boolean createNonExistingParents,
...
-            } else {
-                log.error("Destination folder does not exist and not configured to create");
+        }
+
+        if (StringUtils.isNotEmpty(filePattern)) {
+            FileObject[] children = srcFile.getChildren();
+            for (FileObject child : children) {
+                boolean result;
+                if (child.getType() == FileType.FILE) {
+                    result = moveFileWithPattern(child, destinationFile, filePattern);
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java:278]: If the first file returns false and the next file returns true, won't this be true again?

Comment (sajinieKavindya) [reply src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java:278]: Idea is to track if there happens any issue while moving a file. If we make it true for success scenarios, there is no way we can know that something went wrong earlier.

File: src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java (hunk 1 above)
Comment (sajinieKavindya) [reply src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java:240]: +1

File: src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java (hunk 3 above)
Comment (sajinieKavindya) [reply src/main/java/org/wso2/carbon/connector/operations/MoveFiles.java]: Updated

**Review by arunans23 — APPROVED** [review]

**Pull request merged**